packages/cli/test/unit/util/deploy/create-git-meta.test.ts

```diff
@@ -0,0 +1,147 @@
+import { join } from 'path';
+import fs from 'fs-extra';
+import {
+  createGitMeta,
+  getRepoData,
+  parseRepoUrl,
+} from '../../../../src/util/deploy/create-git-meta';
+import { BitbucketMeta, GitHubMeta, GitLabMeta } from '../../../../src/types';
+
+const fixture = (name: string) =>
+  join(__dirname, '../../../fixtures/unit/create-git-meta', name);
+
+describe('getRepoData', () => {
+  it('gets repo data for no-email', async () => {
+    const configPath = join(fixture('no-email'), 'config');
+    const data = await getRepoData(configPath);
+    expect(data?.provider).toEqual('github');
+    expect(data?.org).toEqual('MatthewStanciu');
+    expect(data?.repo).toEqual('git-test');
+  });
+  it('gets repo data for no-origin', async () => {
+    const configPath = join(fixture('no-origin'), 'config');
+    const data = await getRepoData(configPath);
+    expect(data).toBeUndefined();
+  });
+});
+
+describe('parseRepoUrl', () => {
+  it('should parse github https url', () => {
+    const parsedUrl = parseRepoUrl('https://github.com/vercel/vercel.git');
+    expect(parsedUrl?.provider).toEqual('github');
+    expect(parsedUrl?.org).toEqual('vercel');
+    expect(parsedUrl?.repo).toEqual('vercel');
```

Comment:
As written, if `parseRepoUrl` returned `null`, the error you would see is something like "undefined not equal to github". I like to make sure that the failure messages are as specific as possible to ease debugging.

You can assert that `parsedUrl` is defined first, like `expect(parsedUrl, 'parseRepoUrl()').toBeDefined()`. This will make the failure message mention "parseRepoUrl()" and you'll know that it returned undefined, instead of something to do with the "github" value.

Note: Jest natively doesn't allow custom messages as a second argument to `expect`. This repo has [mattphillips/jest-expect-message](https://github.com/mattphillips/jest-expect-message) installed to allow it.